ci-operator/step-registry/openshift/image-based/upgrade/recipient/gather/lca/openshift-image-based-upgrade-recipient-gather-lca-commands.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set -euo pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+remote_workdir=$(cat ${SHARED_DIR}/remote_workdir)
+instance_ip=$(cat ${SHARED_DIR}/public_address)
+host=$(cat ${SHARED_DIR}/ssh_user)
+ssh_host_ip="$host@$instance_ip"
+RECIPIENT_VM_NAME=$(cat ${SHARED_DIR}/recipient_vm_name)
+recipient_kubeconfig=${remote_workdir}/ib-orchestrate-vm/bip-orchestrate-vm/workdir-${RECIPIENT_VM_NAME}/auth/kubeconfig
+
+echo "Using Host $instance_ip"
+
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-privatekey")
+
+cat <<EOF > ${SHARED_DIR}/gather_recipient_lca.sh
+#!/bin/bash
+set -euo pipefail
+
+# Helper command
+function oc_command() {
+  oc --kubeconfig ${recipient_kubeconfig} \$@
+}
+
+# Setup directories for data
+cd ${remote_workdir}
+gather_dir=./must-gather-lca-${RECIPIENT_VM_NAME}
+gather_dir_extra=./must-gather-lca-${RECIPIENT_VM_NAME}/extra
+mkdir -p \$gather_dir_extra
+
+# Inspect the namespace
+oc_command adm inspect ns/openshift-lifecycle-agent --dest-dir=\$gather_dir
+
+# Get installation service logs and recert summary files
+node="\$(oc_command get nodes -oname)"
+
+oc_command debug \$node -qn openshift-cluster-node-tuning-operator -- chroot /host/ bash -c 'journalctl -u installation-configuration.service' > \$gather_dir_extra/installation-configuration.service.log
+
+oc_command debug \$node -qn openshift-cluster-node-tuning-operator -- chroot /host/ bash -c 'cat /var/tmp/recert-summary.yaml' > \$gather_dir_extra/recert-summary.yaml
```

Comment:
Now that I have some better understanding of what this file contains I'm not sure how valuable it is to pull it out to the artifacts, @eranco74 can we ignore this file and still get an idea of what's happening via the `installation-configuration.service` or some other type of service logs? 

What I'm concerned about is the value add compared to its risk. The logic that generates this file is outside of this repo so if a change was to accidentally expose secrets, we might not know of a while while as this job runs. (correct me if I'm wrong) It seems like all that file contains is the ephemeral certs for the clusters which can only be gotten once a successful upgrade has occurred, so would that be valuable to have after each run since it won't be there when something goes wrong? 

If we just want to make sure the file exists, maybe a better check is to validate that the file was created and it contains data, what do you think?

Reply:
@omertuc maybe it makes sense to have the recert timings / unsensitive information be split to another file to alleviate these concerns. Personally, I believe Recert should not be problematic as a summary if its sanitized but I understand @eggfoobar concern and especially in pipeline context they are valid.

Reply:
recert already has both sensitive summaries and insensitive summaries. The one generated here is the none-sensitive one

Reply:
All private keys and JWTs are redacted from it, it's completely safe to upload as artifact. Recert will ensure it stays safe in the future too

Reply:
Although a recent PR may have caused a regression to expose the pull secret, that's a bug. I will add a fix for it

Reply:
> Although a recent PR may have caused a regression to expose the pull secret, that's a bug. I will add a fix for it

I think this kind of scenario is exactly what @eggfoobar described as being dangerous if its not spotted. It can leak secrets into CI runs. 

Ultimately a risk assessment.

Reply:
>I think this kind of scenario is exactly what @eggfoobar described as being dangerous if its not spotted. It can leak secrets into CI runs.

Bugs can happen anywhere, I'm not sure how that's a risk inherent to this summary.

Reply:
Agreed @omertuc that's a fair statement bugs can happen so I don't want that to be the reason we don't gather data to help debug, but due to the nature of the file the bug probability increases the risk vector, so I want to make sure there is a value to this file being in our pipeline artifacts. If so, we accept the risk and move on.

The thing that I'm not clear on is, if I understand this correctly, this file won't get generated when the upgrade fails, if that's the case then what error state is this file useful for in this pipeline?

Reply:
I'm moving recert summary gather to it's own task as I think validation that he data is obfuscated is a minimum prior to moving it to the artifacts directory. I don't want to scope to creep more than it already has on this PR so the work will be completed on [OCPEDGE-925](https://issues.redhat.com/browse/OCPEDGE-925)